new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -473,12 +525,204 @@ public void testCreateMiiDomainSameDomainUidDiffNS() {
     }
   }
 
+  @Test
+  @Order(4)
+  @DisplayName("Update the sample-app application to version 2")
+  @Slow
+  @MustNotRunInParallel
+  public void testPatchAppV2() {
```

Comment:
I think the new tests in this PR should be added in a new or other existing It class where the domain creation/existence is a prerequisite, then you don't need to order the tests. Here it is assumed that the domain is up and running by ordering the tests where it should have been verified in beforeEach method. This is a late comment in this PR. If everyone agrees, changes can be made in another PR.

Reply:
Thanks, and yes, I would prefer making this change in a different PR.

Reply:
In order for the patching test cases to be fully independent, we need to do three things: 
1) Move those test methods into a different test class;
2) Create a new initial domain in the BeforeAll handler of the new class; instead of using the domain that is created in the first end-to-end test case;
3) At the end of each test case,  rollback what we have done in the method so that the domain goes back to the initial state, to make the state predictable for the next test case.
Both 2) and 3) will increase the total time of the test suite run.